### PR TITLE
Enable ability to assign ontologised Data Use Restriction at the project level. Fixes #1342

### LINKED ISSUE #1342: Add data use ontology module and field to reference it to project schema
As we move toward supporting managed access data we would like to create a module and field to capture a data use ontology term at the project level.

**What should the change/update be?**

### Create data_use_ontology module

A standard ontology module with text, ontology and ontology_label fields with graph restriction being a child of either
data use limitation DUO:0000001 or data use requirements DUO:0000017 (tagging @lauraclarke if she can have any input here)

### Add data_use field to project.json

* Field name: data_use_restriction (?)
* Field description: A brief description of how the data may legally be used
* Field type: array of ontology objects (or should there be just one?)
* Required: no ?
* CV or enum: reference to data_use_ontology module (see above)

If we decide this needs to be a required field moving forward, it would constitute a major schema change

We will also need to request for the Data Use Ontology to be added to the HCAO.

### PR DESCRIPTION
**Pull request by mshadbolt**
### Release notes

For type/project/project schema:
- Added optional data_restrictions field

For module/ontology:
- Added new 'data_use_ontology' module
 
### Reviews requested

- Need 2 Reviewers to approve because this is a minor update
- Last Reviewer to review/approve, merge changes by Thursday, 25th March

### CODE REVIEW

**Comment by mshadbolt** [comment]
note that this won't pass travis until DUO is imported into our ontology

File: json_schema/module/ontology/data_use_ontology.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        "text"
+    ],
+    "title": "Data use ontology",
+    "name": "data_use_ontology",
+    "type": "object",
+    "properties": {
+        "describedBy":  {
+            "description": "The URL reference to the schema.",
+            "type": "string",
+            "pattern" : "^(http|https)://schema\\.(.*?)humancellatlas.org/module/ontology/(([0-9]{1,}\\.[0-9]{1,}\\.[0-9]{1,})|([a-zA-Z]*?))/biological_macromolecule_ontology"
```
Comment (ESapenaVentura) [inline json_schema/module/ontology/data_use_ontology.json:15]: The pattern should end in `data_use_ontology`

**Review by rays22 — APPROVED** [review]
Looks all right to me.

**Review by willrockout — APPROVED** [review]
Looks good to me!

**Pull request merged**